src/inventory/ArmorInventory.php

```diff
@@ -73,4 +80,20 @@ public function setLeggings(Item $leggings) : void{
 	public function setBoots(Item $boots) : void{
 		$this->setItem(self::SLOT_FEET, $boots);
 	}
+
+	private function validate(Inventory $inventory, Item $item, int $slot) : ?TransactionValidationException{
+		if($item instanceof Armor && $item->getArmorSlot() === $slot){
```

Comment:
Is the second part of condition (`&& $item->getArmorSlot() === $slot`) needed? If I read code correctly, the second condition (`if($item->getArmorSlot() !== $slot){`) will be always false since it’s already assumed that the item armor slot should equals the inventory slot